pkg/activator/handler/concurrency_reporter.go

```diff
@@ -61,7 +73,104 @@ func NewConcurrencyReporter(ctx context.Context, podName string,
 		reqCh:   reqCh,
 		statCh:  statCh,
 		rl:      revisioninformer.Get(ctx).Lister(),
+
+		stats:                make(map[types.NamespacedName]*network.RequestStats),
+		reportedFirstRequest: make(map[types.NamespacedName]float64),
+	}
+}
+
+// handleEvent handles request events (in, out) and updates the respective stats.
+func (cr *ConcurrencyReporter) handleEvent(event network.ReqEvent) {
+	stats, msgs := cr.getOrCreateStat(event)
+	if len(msgs) > 0 {
+		cr.statCh <- msgs
+	}
+	stats.HandleEvent(event)
+}
+
+// getOrCreateStat gets a stat from the state if present.
+// If absent it creates a new one and returns it, potentially returning a StatMessage too
+// to trigger an immediate scale-from-0.
+func (cr *ConcurrencyReporter) getOrCreateStat(event network.ReqEvent) (*network.RequestStats, []asmetrics.StatMessage) {
```

Comment:
why would this return a slice?

Reply:
Kind of just an artifact of existing code. I guess we can make it a pointer.

Reply:
or just a value and let compiler make it a pointer if needed